core-api/src/worker.rs

```diff
@@ -218,18 +216,31 @@ impl WorkerConfigBuilder {
             return Err("max_outstanding_* fields are mutually exclusive with `tuner`".to_owned());
         }
 
-        if self.use_worker_versioning.unwrap_or_default()
-            && self
-                .worker_build_id
-                .as_ref()
-                .map(|s| s.is_empty())
-                .unwrap_or_default()
-        {
-            return Err(
-                "`worker_build_id` must be non-empty when `use_worker_versioning` is true"
-                    .to_owned(),
-            );
+        if let Some(wv) = self.versioning_strategy.as_ref() {
+            match wv {
+                WorkerVersioningStrategy::None { .. } => {}
+                WorkerVersioningStrategy::WorkerDeploymentBased(d) => {
+                    if d.use_worker_versioning
+                        && (d.version.build_id.is_empty() || d.version.deployment_name.is_empty())
+                    {
+                        return Err(
+                            "WorkerDeploymentVersion must have a non-empty build_id and \
+                     deployment_name when deployment based versioning is enabled"
+                                .to_owned(),
+                        );
+                    }
+                }
+                WorkerVersioningStrategy::LegcayBuildIdBased { build_id } => {
+                    if build_id.is_empty() {
+                        return Err(
+                            "Legacy build id based versioning must have a non-empty build_id"
```

Comment:
Haha but I never know how to do this

```suggestion
                            "Legacy build id-based versioning must have a non-empty build_id"
```

Reply:
How to do what?

Reply:
Use a hyphen to join a pair of space-separated words with a third word.